api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -74,5 +85,23 @@
    *
    * @return this for method chaining
    */
-  OverwriteFiles validateAddedFiles();
+  OverwriteFiles validateAddedFilesMatchOverwriteFilter();
+
+  /**
+   * Enables validation that files added concurrently do not conflict with this commit's operation.
+   * <p>
+   * This method should be called when the table is queried to determine which files to delete/append.
+   * If a concurrent operation commits a new file after the data was read and that file might
+   * contain rows matching the specified conflict detection filter, the overwrite operation
+   * will detect this during retries and fail.
+   * <p>
+   * Calling this method with a correct conflict detection filter is required to maintain
+   * serializable isolation for eager update/delete operations. Otherwise, the isolation level
+   * will be snapshot isolation.
+   *
+   * @param readSnapshotId the snapshot id that was used to read the data
```

Comment:
This should document what happens when `readSnapshotId` is null since that's supported.